### PR TITLE
Add tlsCertificate and certificateValidationContext to XdsAttributes

### PR DESCRIPTION
**Pull request by sanjaypujare**
Needed for passing those values to the Sds component (channel and server builders)
Also address comments in a previous PR

### CODE REVIEW

File: xds/src/main/java/io/grpc/xds/XdsAttributes.java [hunk 1]
```diff
@@ -32,5 +34,21 @@
   public static final Attributes.Key<SdsSecretConfig> ATTR_SDS_CONFIG =
           Attributes.Key.create("io.grpc.xds.XdsAttributes.sdsSecretConfig");
 
+  /**
+   * Attribute key for TlsCertificate of a subchannel.
+   *
```
Comment (creamsoup) [inline xds/src/main/java/io/grpc/xds/XdsAttributes.java]: nit: those extra lines bother me. can you remove those (L39, L47, L27?)

File: xds/src/main/java/io/grpc/xds/sds/SecretProvider.java [hunk 2]
```diff
@@ -23,8 +23,7 @@
  * A SecretProvider is a "container" or provider of a secret. This is used by gRPC-xds to access
  * secrets, so is not part of the public API of gRPC. This "container" may represent a stream that
  * is receiving the requested secret(s) or it could represent file-system based secret(s) that are
- * dynamic. Synchronous and Asynchronous methods to access the underlying secret are available. See
- * {@link SecretManager} for a note on lifecycle management
+ * dynamic.
```
Comment (creamsoup) [inline xds/src/main/java/io/grpc/xds/sds/SecretProvider.java]: this is unrelated changes, probably need to be a separate PR. (or 2 commits and rebase the change)

File: xds/src/main/java/io/grpc/xds/XdsAttributes.java (hunk 1 above)
Comment (sanjaypujare) [reply xds/src/main/java/io/grpc/xds/XdsAttributes.java]: done

File: xds/src/main/java/io/grpc/xds/sds/SecretProvider.java (hunk 2 above)
Comment (sanjaypujare) [reply xds/src/main/java/io/grpc/xds/sds/SecretProvider.java]: done
